docs/configuration.md

```diff
@@ -299,6 +299,47 @@ By default, Ruff will also skip any files that are omitted via `.ignore`, `.giti
 Files that are passed to `ruff` directly are always analyzed, regardless of the above criteria.
 For example, `ruff check /path/to/excluded/file.py` will always lint `file.py`.
 
+### Default inclusions
+
+By default, Ruff will discover files matching `*.py`, `*.ipy`, or `pyproject.toml`. 
+
+To lint or format files with additional file extensions, use the [`extend-include`](settings.md#extend-include) setting.
+
+=== "pyproject.toml"
+
+    ```toml
+    [tool.ruff.lint]
+    extend-include = ["*.ipynb"]
+    ```
+
+=== "ruff.toml"
+
+    ```toml
+    [lint]
+    extend-include = ["*.ipynb"]
+    ```
+
+You can also change the default selection using the [`include`](settings.md#include) setting.
+
+
+=== "pyproject.toml"
+
+    ```toml
+    [tool.ruff.lint]
+    include = ["pyproject.toml", "src/**/*.py", "scripts/**/*.py"]
+    ```
+
+=== "ruff.toml"
+
+    ```toml
+    [lint]
+    include = ["pyproject.toml", "src/**/*.py", "scripts/**/*.py"]
+    ```
+
+!!! warning
+    Paths provided to `include` _must_ match files. For example, `include = ["src"]` will fail since it
+matches a directory.
```

Comment:
This really feels like it should not be the case. We should join our default inclusions to directories. I'd prefer to tackle that separately though.

Reply:
What would be examples of use-cases in which one would want to include an entire directory?

Reply:
As in #3970 e.g. `ruff check` with `include = ["src", "tests"]`

Reply:
> As in #3970 e.g. `ruff check` with `include = ["src", "tests"]`

Is that means cli also will inherit `include` behavior (globs path)?